src/librustc/mir/interpret/mod.rs

```diff
@@ -440,7 +488,7 @@ pub enum AllocType<'tcx, M> {
 
 pub struct AllocMap<'tcx, M> {
     /// Lets you know what an AllocId refers to
-    id_to_type: FxHashMap<AllocId, AllocType<'tcx, M>>,
+    id_to_type: FxHashMap<u64, AllocType<'tcx, M>>,
```

Comment:
Now we lose the newtype protection for not mixing up IDs.

I don't understand the purpose of this entire commit, at all.